src/main/java/net/consensys/orion/network/NodeHttpClientBuilder.java

```diff
@@ -37,13 +38,18 @@ public static HttpClient build(Vertx vertx, Config config, int clientTimeoutMs)
 
       PemKeyCertOptions pemKeyCertOptions =
           new PemKeyCertOptions().setKeyPath(tlsClientKey.toString()).setCertPath(tlsClientCert.toString());
-      for (Path chainCert : config.tlsClientChain()) {
-        pemKeyCertOptions.addCertPath(chainCert.toAbsolutePath().toString());
-      }
 
       options.setSsl(true);
       options.setPemKeyCertOptions(pemKeyCertOptions);
 
+      if (!config.tlsClientChain().isEmpty()) {
+        PemTrustOptions pemTrustOptions = new PemTrustOptions();
+        for (Path chainCert : config.tlsClientChain()) {
+          pemTrustOptions.addCertPath(chainCert.toAbsolutePath().toString());
+        }
+        options.setPemTrustOptions(pemTrustOptions);
+      }
+
```

Comment:
nit: now that it's more than two lines it looks like dupe

Reply:
It is using Vertx builders. I don't think we'll get a lot from abstracting it into a separate class atm.